### DISCUSSION TITLE
Concern and clarity needed regarding restrictive language attached to the Apache 2.0 License

### DISCUSSION THREAD

**Discussion by mclean-connor** [FlowiseAI/Flowise#1459] [General]
November 23, 2023, FlowiseAI appended language to the Apache 2.0 license that we all operate under. Specifically: 
```
Flowise is governed by the Apache License 2.0, with additional terms and conditions outlined below:

Flowise can be used for commercial purposes for "backend-as-a-service" for your applications or as a development platform for enterprises. However, under specific conditions, you must reach out to the project's administrators to secure a commercial license:

a. Multi-tenant SaaS service: Unless you have explicit written authorization from Flowise, you may not utilize the Flowise source code to operate a multi-tenant SaaS service that closely resembles the Flowise cloud-based services. b. Logo and copyright information: While using Flowise in commercial application, you are prohibited from removing or altering the LOGO or copyright information displayed in the Flowise console and UI.

For inquiries regarding licensing matters, please contact [hello@flowiseai.com](mailto:hello@flowiseai.com) via email.

Contributors are required to consent to the following terms related to their contributed code:

a. The project maintainers have the authority to modify the open-source agreement to be more stringent or lenient. b. Contributed code can be used for commercial purposes, including Flowise's cloud-based services.

All other rights and restrictions are in accordance with the Apache License 2.0.
```

The additional terms and conditions added to the Apache License 2.0 raise important issues regarding compatibility with the original Apache License 2.0. The Apache License is a permissive free software license written by the Apache Software Foundation (ASF). It allows users to use the software for any purpose, to distribute it, to modify it, and to distribute modified versions of the software under the terms of the license, without concern for royalties and/or ownership.

Below, key points of concern pertaining to the added terms:

Restriction on Use for Multi-tenant SaaS Service: The original Apache 2.0 license does not restrict the type of commercial use. Adding a clause that prohibits the use of Flowise community-contributed source code to operate a multi-tenant SaaS service without explicit authorization is a significant restriction that is not in line with the freedoms granted by the Apache 2.0 license.

Requirement to Maintain Logo and Copyright Information: While the Apache License 2.0 requires that you must cause any modified files to carry prominent notices stating that you changed the files, it does not require the original logo or copyright information to be maintained in a user interface. This added restriction goes beyond what the Apache License stipulates. Actually, logos are traditionally remove under Apache 2.0 since, under Apache 2.0, you are not granted permission to use them in a way that suggests endorsement by the original creator.

Authority to Modify the Open-source Agreement: The Apache License 2.0 is a fixed agreement and does not provide for unilateral modifications by project maintainers. The terms recently introduced undermine the stability and predictability of the licensing terms that Apache 2.0 makes clear and secure. 

To summarize. These additional terms impose restrictions that are not present in the Apache License 2.0 and are considered incompatible with the principles of the license. The changes FlowiseAI made are not consistent with the current Apache 2.0 license. This will lead to legal ambiguity and potential challenges, especially in terms of how these additional terms are enforced and interpreted relating to the freedoms granted by the Apache 2.0 License.

It's important for open source projects to ensure that any additional terms or agreements are fully compatible with the licenses they choose to adopt. As an open source community, I know all contributors to this project would like to see these values upheld, as work to produce this open source software has not been completed by FlowiseAI alone. 

If FlowiseAI wishes to incorporate restrictive language or additional enterprise/commercial licensing, then a separate repo or directory within Flowise should be created with its own licensing and terms that apply. 

Please elaborate on the recently introduced terminology/language and respond to the full content provided above. I am confident that the vast community of Flowise users and, notably, the contributors, would greatly value enhanced transparency. Thank you!

**Comment by BigFoxMedia** [comment]
Following

**Comment by HenryHengZJ** [comment]
We have removed the restrictions. Reverted back to Apache 2.0
